server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java

```diff
@@ -14,57 +14,49 @@
  */
 package org.eclipse.lsp.cobol.core.model.tree;
 
+import com.google.common.collect.ArrayListMultimap;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.ToString;
-import org.eclipse.lsp.cobol.core.messages.MessageService;
+import lombok.extern.slf4j.Slf4j;
 import org.eclipse.lsp.cobol.core.messages.MessageTemplate;
 import org.eclipse.lsp.cobol.core.model.ErrorSeverity;
 import org.eclipse.lsp.cobol.core.model.Locality;
 import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.model.VariableUsageUtils;
 import org.eclipse.lsp.cobol.core.model.tree.statements.StatementNode;
+import org.eclipse.lsp.cobol.core.model.tree.variables.MnemonicNameNode;
 import org.eclipse.lsp.cobol.core.model.tree.variables.VariableNode;
+import org.eclipse.lsp.cobol.core.model.tree.variables.VariableUsageNode;
 import org.eclipse.lsp.cobol.core.model.variables.NodeConverter;
 import org.eclipse.lsp.cobol.core.model.variables.Variable;
-import org.eclipse.lsp.cobol.core.visitor.VariableUsageDelegate;
+import org.eclipse.lsp.cobol.core.semantics.PredefinedVariables;
 
 import java.util.*;
 
 import static java.util.stream.Collectors.toList;
-import static java.util.stream.Collectors.toSet;
-import static org.eclipse.lsp.cobol.core.model.tree.NodeType.PROGRAM;
-import static org.eclipse.lsp.cobol.core.model.tree.NodeType.STATEMENT;
+import static org.eclipse.lsp.cobol.core.model.tree.NodeType.*;
+import static org.eclipse.lsp.cobol.core.semantics.PredefinedVariables.PREDEFINED;
+import static org.eclipse.lsp.cobol.service.CopybookServiceImpl.PREF_IMPLICIT;
 
 /** This class represents program context in COBOL. */
 @ToString(callSuper = true)
 @Getter
 @EqualsAndHashCode(callSuper = true)
+@Slf4j
 public class ProgramNode extends Node {
+  private final NodeConverter nodeConverter = new NodeConverter();
```

Comment:
Why don't you exclude it anymore?

Reply:
Because we still need to convert VariableNode to Variable... AnalisysResult and the rest still depend on Variable.

Reply:
I mean, you had `@EqualsAndHashCode.Exclude` before.